File: arrays/longest_consecutive_streak/longest_consecutive.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
  const uniques = new Set(nums);
  let longest = 0;

  for (const num of uniques) {
      if (!uniques.has(num-1)) {
          length = 1;
          while (uniques.has(num+length)) {
              length++;
          }
          if (length > longest) longest = length;
      }
  }
  return longest;
};

File: arrays/longest_consecutive_streak/longest_consecutive.py
class Solution(object):
    def longestConsecutive(self, nums):
        uniques = set(nums)
        longest = 0

        for num in uniques:
            if not (num-1) in uniques:
                length = 1
                while (num+length) in uniques:
                    length+=1
                longest=max(length,longest)
        return longest